File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) 
    {
        if(c<=1)
        return 1;
        long long int start =0,end = sqrt(c);
        int flag =0;
        while(start<=end)
        {
            long long int val = start*start + end*end;
            val == c ? flag =1 : val<c ? start++ : end--; 
           if(flag)
           return 1;
        }    
        return 0;
    }
};

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution(object):
    def judgeSquareSum(self, c):
        
        if(c == 0):
            return True

        for i in range(1,int(sqrt(c))+1):
            a = i
            b = c - (i**2)

            sqrtb = int(sqrt(b)) 

            if(sqrtb**2 == b):
                return True
        return False
        